### ISSUE TITLE
More configurability

### ISSUE THREAD

**Issue by Manishearth** [rust-fuzz/cargo-fuzz#3]
Add the ability to choose between debug/release. Pass down any other useful rustc flags.

**Comment by ruuda** [comment]
Being able to pass arguments to libfuzzer, such as `-max_size=` or the location of the corpus, would also be useful. Currently I run `cargo fuzz --fuzz-target fuzzer_script_1` to trigger the build, kill it, and then I start `fuzz/target/debug/fuzzer_script_1` with the desired flags.

Simply passing on any extra arguments to the target program would be sufficient.

**Comment by Manishearth** [comment]
Yep. `cargo fuzz --fuzz-target foo -- FUZZ ARGS BLAH BLAH` is what we'd probably use.

**Comment by nagisa** [comment]
You can pass args to the underlying binary now

(try `cargo fuzz run target -- -help=1`)

It is still desirable to have more configuration for how the fuzz target is compiled (release/debug, debug-assertions on or off, etc)

**Comment by Bobo1239** [comment]
Just asking for confirmation: Ignoring a function/crate (like https://clang.llvm.org/docs/SanitizerSpecialCaseList.html) would need rustc support, right?
(In my case [rayon](https://github.com/nikomatsakis/rayon) is triggering ASan though I believe that is https://github.com/rust-lang/rust/issues/39882.)

**Comment by nagisa** [comment]
No, not really. It likely can be expressed through `-C llvm-args` in `RUSTFLAGS`.

**Comment by Bobo1239** [comment]
I did try `RUSTFLAGS="-C llvm-args=-fsanitize-blacklist=blacklist.txt" cargo fuzz run fuzzer_script_1` but that only results in
```
error: failed to run `rustc` to learn about target-specific information
Caused by:
  process didn't exit successfully: `rustc - --crate-name ___ --print=file-names -C llvm-args=-fsanitize-blacklist=blacklist.txt -Cpasses=sancov -Cllvm-args=-sanitizer-coverage-level=3 -Zsanitizer=address -Cpanic=abort --crate-type bin --crate-type dylib --crate-type proc-macro --crate-type rlib --target x86_64-unknown-linux-gnu` (exit code: 1)
--- stderr
rustc: Unknown command line argument '-fsanitize-blacklist=blacklist.txt'.  Try: 'rustc -help'
rustc: Did you mean '-dfsan-abilist=blacklist.txt'?
```
Research led me to [this commit](https://github.com/llvm-mirror/llvm/commit/f78f9fda313fcb035f19e5760c2483bd1161fb08) hence my assumption.
